### DISCUSSION TITLE
Meaning of ``int bcfcomp`` in ``FillPatchSingleLevel()`` and ``int cbccomp``, ``int 	fbccomp``, and ``int bcscomp`` in ``FillPatchTwoLevels()``

### DISCUSSION THREAD

**Discussion by ZiyangHuang** [AMReX-Codes/amrex#3929] [Q&A] [answered]
Hi!

I am trying to call ``FillPatchSingleLevel()`` and ``FillPatchTwoLevels()`` to fill a ``Multifab mf`` using data ``Vector<MultiFab> dof_new`` and ``Vector<Multifab> dof_old``. Both ``dof_new`` and ``dof_old`` have ``Ncomp`` components. I also defined ``Vector<BCRec> bcs(Ncomp)`` storing the boundary conditions for individual components. 

Right now, I am passing ``int scomp`` to ``int bcfcomp`` in ``FillPatchSingleLevel()`` and ``int cbccomp``, ``int fbccomp``, and ``int bcscomp`` in ``FillPatchTwoLevels()``, but I am not sure if it is correct.

Thanks!

**Comment by cgilet** [comment]
It depends on what you're using for `cbc` and `fbc`. `cbccomp` will be the first component used within `cbc`, and `fbccomp` will be the first component used within `fbc`. You need to make sure all these components point to the same "thing", eg. x-velocity.

**Reply by WeiqunZhang** [reply, marked as answer]
Adding to what @cgilet said, `bcfcomp` will be passed to the `physbcf` argument. If your `physbcf` is `amrex::PhysBCFunct` and you have used `Vector<BCRec> bcs(Ncomp)` to create it, then amrex will assume that `bcs[bcfcomp+icomp]` is the boundary type for component `dcomp+icomp` of the destination multifab (i.e., `mf`) and component `scomp+icomp` of the source multifab (i.e., `smf`), where `icomp` is in `[0,ncomp)`.

**Reply by ZiyangHuang** [reply]
Thanks for your replies. It is clear for me now.
